mitodl/open-discussions
# Implement adding and editing widgets

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #1508 

#### What's this PR do?
Implements adding and editing widgets, similar to how it's done in the video. The textboxes look different and preview is not implemented in this PR.

#### How should this be manually tested?
Click the gear icon then click 'manage widgets'. You should see '+ Add widget'. Click that and go through the dialog. If you click cancel at any point the dialog should close and nothing should change on the widget list.

If you fill out the items and click 'Create widget' it should add the widget to the widget list. Note that the widget is not saved to the server yet. Click 'Done' and you should see the widget list refresh with the new widget in place, rendering its contents.

Click 'manage widgets' again and click the edit button next to a widget. You should see a dialog which shows the fields for that widget plus a field for the widget title. If you click cancel no changes should be shown in the widget list. If you click 'Edit widget' it should update the widget list with your changes, but nothing should be saved to the server yet until you click 'Done'.

You should be able to delete or move widgets around without any problems while editing or adding widgets.

#### Screenshots
Click "+ Add widget", then clicking next before selecting a widget type:
![select_widget_1](https://user-images.githubusercontent.com/863262/50365148-3adbb080-0541-11e9-8337-27aef5df04bc.png)

Clicking next before filling out the widget title:
![screenshot from 2018-12-21 16-54-08](https://user-images.githubusercontent.com/863262/50365173-4dee8080-0541-11e9-95f9-8c7da0826d20.png)

Edit widget:
![screenshot from 2018-12-21 16-54-49](https://user-images.githubusercontent.com/863262/50365276-b50c3500-0541-11e9-919f-9ded9988d99c.png)

No mobile screenshots because #1518 is not done
